src/test/java/plugins/KerberosSsoTest.java

```diff
@@ -200,14 +203,29 @@ private FirefoxDriver getNegotiatingFirefox(KerberosContainer kdc, String tokenC
         FirefoxProfile profile = new FirefoxProfile();
         profile.setAlwaysLoadNoFocusLib(true);
         // Allow auth negotiation for jenkins under test
-        profile.setPreference("network.negotiate-auth.trusted-uris", jenkins.url.toExternalForm());
-        profile.setPreference("network.negotiate-auth.delegation-uris", jenkins.url.toExternalForm());
+        String trustedUris = jenkins.url.toExternalForm();
+        String jenkins_local_hostname = System.getenv("JENKINS_LOCAL_HOSTNAME");
+        // if JENKINS_LOCAL_HOSTNAME is set, we add this to FF nego uris
+        if (jenkins_local_hostname != null && !jenkins_local_hostname.isEmpty()) {
+            try {
+                // In the case where JENKINS_LOCAL_HOSTNAME is an IP,
+                // we need to add its resolved hostname for auth negociation
+                String hostName = InetAddress.getByName(jenkins_local_hostname).getCanonicalHostName();
+                trustedUris = trustedUris + ", " + hostName;
```

Comment:
That is what I saw in the krb5 trace. It was trying to resolve hostname and then domain name if an IP address was used